api/bazel/repository_locations.bzl

```diff
@@ -32,9 +32,9 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Bazel build tools",
         project_desc = "Developer tools for working with Google's bazel buildtool.",
         project_url = "https://github.com/bazelbuild/buildtools",
-        version = "4.0.1",
-        sha256 = "c28eef4d30ba1a195c6837acf6c75a4034981f5b4002dda3c5aa6e48ce023cf1",
-        release_date = "2021-03-01",
+        version = "4.2.1",
+        sha256 = "b8b69615e8d9ade79f3612311b8d0c4dfe01017420c90eed11db15e9e7c9ff3c",
+        release_date = "2021-09-10",
```

Comment:
As per CI failure the UTC for this release is `2021-09-11`. The way I find this date is right click on tagged date text on Github, inspect the element in either Firefox or Chrome and find the `datetime` attribute below which is the UTC of the release. This is the date the checkers use

```html
<relative-time datetime="2021-09-11T01:38:17Z" class="no-wrap" title="Sep 10, 2021, 6:38 PM PDT">26 days ago</relative-time>
```